### ISSUE TITLE
`StoryblokStory` and `StoryblokServerComponent` don't support passing down arbitrary props anymore

### ISSUE THREAD

**Issue by TomRadford** [storyblok/storyblok-react#1302]
### Describe the issue you're facing

In `"@storyblok/react": "4.0.0"` in NextJS 15 we were able to pass down arbitary props down to pages/blocks like so:
```ts
  <StoryblokServerComponent // or StoryblokStory
            blok={footerData.story.content}
            // any prop what will get passed down:
            backgroundVariant={data.story.content.background_variant}
            browser={browser}
            {...anyOtherProp}
/>
```

in 4.2.1 this yields TS errors:
<img width="894" alt="image" src="https://github.com/user-attachments/assets/f51cbfef-0519-4ff8-8fb4-87ae5af28309" />

### Reproduction

.

### System Info

```shell
"react": "19.0.0",
"react-dom": "19.0.0",
"@storyblok/react": "^4.2.1",
"next": "15.1.0",

  System:
    OS: macOS 14.3
    CPU: (8) arm64 Apple M3
    Memory: 87.33 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.10.0 - ~/.local/state/fnm_multishells/54467_1734429279113/bin/node
    npm: 10.9.0 - ~/.local/state/fnm_multishells/54467_1734429279113/bin/npm
    pnpm: 9.12.3 - ~/.local/state/fnm_multishells/54467_1734429279113/bin/pnpm
  Browsers:
    Chrome: 131.0.6778.140
    Safari: 17.3
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://www.storyblok.com/trust-center#code-of-conduct?utm_source=github.com&utm_medium=readme&utm_campaign=storyblok-js-client)

**Comment by edodusi** [comment]
@TomRadford we have just fixed this in 4.3.1 (#1296)

**Comment by TomRadford** [comment]
Amazing! Thanks @edodusi ✨
